publish-image/action.yml

```
@@ -109,9 +109,20 @@ runs:
         # Extract the digest from the image repo digest (right side of '@')
         DIGEST=${IMAGE_REPO_DIGEST#*@}
 
+        # URL encode the digest and image repository, needed for the purl
+        URLENCODED_DIGEST=$(echo "$DIGEST" | jq -Rr @uri)
+        URLENCODED_IMAGE_REPOSITORY=$(echo "$IMAGE_REPOSITORY" | jq -Rr @uri)
```

suggestion: remove the `echo` and pipe.

```suggestion
        URLENCODED_DIGEST=$(jq -rn --arg input "$DIGEST" '$input | @uri')
        URLENCODED_IMAGE_REPOSITORY=$(jq -rn --arg input "$IMAGE_REPOSITORY" '$input | @uri')
```

---

Done